frontend/expr_inferrer.py

```diff
@@ -375,33 +376,69 @@ def _get_args_types(args, context, instance, solver):
     return args_types
 
 
-def _infer_annotated_function_call(args_types, solver, annotations):
-    result_type = solver.new_z3_const("call")
-    solver.annotation_resolver.add_annotated_function_axioms(args_types, solver, annotations, result_type)
+def _infer_annotated_function_call(args_types, solver, annotations, result_type):
+    return solver.annotation_resolver.get_annotated_function_axioms(args_types, solver, annotations, result_type)
 
-    return result_type
+
+def _get_builtin_method_call_axioms(args_types, solver, context, result_type, method_name):
+    """Get the axioms of built-in method calls"""
+    possible_methods = context.get_matching_methods(method_name)
+    method_axioms = []
+    for method in possible_methods:
+        cur_method_axioms = _infer_annotated_function_call(args_types, solver, method, result_type)
+        if cur_method_axioms is not None:
+            method_axioms.append(cur_method_axioms)
+    return method_axioms
 
 
 def infer_func_call(node, context, solver):
     """Infer the type of a function call, and unify the call types with the function parameters"""
     instance = None
     result_type = solver.new_z3_const("call")
+    call_axioms = []
     if isinstance(node.func, ast.Attribute):
-        called, instance = infer(node.func, context, solver, True)
+        # Add axioms for built-in methods
+        instance = infer(node.func.value, context, solver)
+        if isinstance(instance, Context):
+            # Module access; instance is a module, so don't add it as a receiver to `arg_types`
+            instance = None
         args_types = _get_args_types(node.args, context, instance, solver)
+        call_axioms += _get_builtin_method_call_axioms(args_types, solver, context, result_type, node.func.attr)
     else:
         args_types = _get_args_types(node.args, context, instance, solver)
-        called = infer(node.func, context, solver)
+
+    called = infer(node.func, context, solver)
```

Comment:
Yes intentional.

We cannot move arg_types statement because `arg_types` maybe coming from the `then` branch, not the `else` one.

Basically the then branch adds the axioms for the built-in method calls, and then later on axioms for normal calls are added. However, if it's an attribute-call, the `arg_types` would have already been inferred before, and there's no need to re-infer them.

I'll try to re-write it too in a clearer way.